simulator/host_datastore_system.go

```diff
@@ -83,11 +84,28 @@ func (dss *HostDatastoreSystem) add(ctx *Context, ds *Datastore) *soap.Fault {
 	parent := hostParent(dss.Host)
 	Map.AddReference(ctx, parent, &parent.Datastore, ds.Self)
 
-	browser := &HostDatastoreBrowser{}
-	browser.Datastore = dss.Datastore
-	ds.Browser = Map.Put(browser).Reference()
+	if Map.Get(ds.Self) == nil {
+		browser := &HostDatastoreBrowser{}
+		browser.Datastore = dss.Datastore
+		ds.Browser = Map.Put(browser).Reference()
 
-	folderPutChild(ctx, folder, ds)
+		var capacity units.ByteSize
+		b := os.Getenv("VCSIM_DATASTORE_CAPACITY")
+		if b == "" {
+			b = "1TB"
+		}
+		capacity.Set(b)
```

Comment:
Ignoring error is fine here?

Reply:
Ah yeah, we should just `panic()` on error here I think.